MapleServer2/Types/Gemstone.cs

```diff
@@ -108,7 +111,7 @@ public void CreateSockets(Item parent)
     public void OpenSockets()
     {
         // roll to unlock sockets
-        for (int i = 0; i < Sockets.Count; i++)
+        foreach (GemSocket socket in Sockets)
         {
             int successNumber = Random.Shared.Next(0, 100);
 
```

Comment:
Was this tested? I remember specifically using a for loop instead of foreach loop for a reason.

Reply:
No but how would it make a difference?